gammapy/datasets/simulate.py

```diff
@@ -576,3 +578,99 @@ def run(self, dataset, observation=None):
         geom = dataset._geom
         selection = geom.contains(events.map_coord(geom))
         return events.select_row_subset(selection)
+
+
+class SimulatedObservationMaker(MapDatasetEventSampler):
+    """
+    Sample event lists for a given observation.
+    """
+
+    def __init__(
+        self,
+        energy_axis_true,
+        energy_axis,
+        migra_axis,
+        map_width,
+        map_bin_size=0.02 * u.deg,
+        random_state="random-seed",
+    ):
+        self.random_state = get_random_state(random_state)
+        self.energy_axis_true = energy_axis_true
+        self.energy_axis = energy_axis
+        self.migra_axis = migra_axis
+        self.map_width = map_width
+        self.bin_size = map_bin_size
+
+    def run(self, observation: Observation, models=None):
+        """Run the event sampler, applying IRF corrections.
+
+        Parameters
+        ----------
+        observation : `~gammapy.data.Observation`
+            Observation to be simulated.
+        models : `~gammapy.modeling.Models`
+
+        Returns
+        -------
+        events : `~gammapy.data.EventList`
+            Event list.
+        """
+        if models is None:
+            models = Models()
+
+        pointing_icrs = observation.pointing.get_icrs(
```

Comment:
Since we explicitly only support fixed pointing in ra/dec, I think this should check and directly use `fixed_icrs` instead of asking for the icrs at mid of run time.